### PR TITLE
adding code and updated docs/readme

### PR DESCRIPTION
**Pull request by BlitzCityDIY**
Adding library, example and docs/readme.

### CODE REVIEW

**Comment by BlitzCityDIY** [comment]
in thinking about what limor said in the internal meeting yesterday about enums, i'm wondering if i should property-ify a few of the parameter-setters, rather than having them just be RWBits, so that we can check for correct values

File: adafruit_vcnl4020.py [hunk 1]
```diff
@@ -16,22 +20,239 @@
...
+        return (self._command_reg >> 6) & 0x01
+
+    @property
+    def proximity_ready(self) -> int:
+        """
+        Status of proximity data.
+
+        :return: True if proximity data is ready, otherwise false.
+        """
+        return (self._command_reg >> 5) & 0x01
```
Comment (tannewt) [inline adafruit_vcnl4020.py:205]: This is what a ROBit will do. You can use that instead.

File: adafruit_vcnl4020.py [hunk 2]
```diff
@@ -16,22 +20,239 @@
...
+        """
+        Continuous conversion mode for ambient light measurement.
+
+        :return: True if enabled, False if disabled.
+        """
+        return bool(self._continuous_conversion_bit)
+
+    @continuous_conversion.setter
+    def continuous_conversion(self, enable: bool) -> None:
+        self._continuous_conversion_bit = enable
```
Comment (tannewt) [inline adafruit_vcnl4020.py:258]: This is also what ROBit does. It gets and sets a single bit based on a bool. https://github.com/adafruit/Adafruit_CircuitPython_Register/blob/main/adafruit_register/i2c_bit.py

**Review by tannewt — CHANGES_REQUESTED** [review]
I wouldn't use ENUM-like things because the variable names are pretty large in memory. Instead, I'd do the list of valid numbers approach for PROX_RATE, AMBIENT_RATE and PROX_FREQ. So the properties would just take in a number and validate against the list. AVG_SAMPLES and INT_COUNT can also take in an int and verify it is a power of two between the valid range.

The property doc can explain the valid numeric values and the units.

**Comment by BlitzCityDIY** [comment]
this PR has been acting funky since friday and isn't showing my recent commit, despite a few attempts at pushing. going to close and reopen to see if that pushes it

**Pull request closed without merging**